Minio/ApiEndpoints/BucketOperations.cs

```diff
@@ -171,35 +179,30 @@ public IObservable<Item> ListObjectsAsync(string bucketName, string prefix = nul
         /// Gets the list of objects in the bucket filtered by prefix
         /// </summary>
         /// <param name="bucketName">Bucket to list objects from</param>
-        /// <param name="prefix">Filters all objects not beginning with a given prefix</param>
-        /// <param name="recursive">Set to false to emulate a directory</param>
+        /// <param name="prefix">Filters all objects starting with a given prefix</param>
+        /// <param name="delimiter">Delimit the output upto this character</param>
         /// <param name="marker">marks location in the iterator sequence</param>
         /// <returns>Task with a tuple populated with objects</returns>
         /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
 
-        private async Task<Tuple<ListBucketResult, List<Item>>> GetObjectListAsync(string bucketName, string prefix, bool recursive, string marker, CancellationToken cancellationToken = default(CancellationToken))
+        private async Task<Tuple<ListBucketResult, List<Item>>> GetObjectListAsync(string bucketName, string prefix, string delimiter, string marker, CancellationToken cancellationToken = default(CancellationToken))
         {
             var queries = new List<string>();
-            if (!recursive)
-            {
-                queries.Add("delimiter=%2F");
-            }
-            if (prefix != null)
-            {
-                queries.Add("prefix=" + Uri.EscapeDataString(prefix));
-            }
+            queries.Add("delimiter="+ Uri.EscapeDataString(delimiter));
```

Comment:
My thesis was that they shouldn't be null in the first place, they are simply empty values. So there is no API breakage and it also makes it unnecessary for users to now pass empty string explicitly, and not throw exception for null values.

Reply:
When user passes `null` for `prefix`, with this PR `Uri.EscapeDataString` would throw `ArgumentNullException`, right? What should our library do in this case?

Reply:
That is a possibility but then the exception thrown isn't it appropriate ?..

Are you saying we should handle both null and empty ? Isn't that a bit too much.. setting prefix or bucket as null is clearly a bug in the application.

Reply:
Yes, passing `null` is an application bug. I am thinking if we rethrow it as an exception type defined in our library, applications can know the set of exceptions that our library throws. Without which applications need to learn the possible set of exceptions from incorrect use or reading our source code.

Reply:
Generally I thought that cannot be controlled because there are many other exceptions too. I can add it back though and certainly eat the exception and handle it.

Let me talk to @poornas as well for her inputs. If it's meant to handle it then null and empty we can treat as same.

Reply:
It seems appropriate to treat null as empty string here.The API signature defaults to  null for unspecified prefix and delimiter.